Modules/socketmodule.c

```diff
@@ -1579,6 +1582,31 @@ makesockaddr(SOCKET_T sockfd, struct sockaddr *addr, size_t addrlen, int proto)
     }
 #endif /* HAVE_SOCKADDR_ALG */
 
+#ifdef AF_HYPERV
+    case AF_HYPERV:
+    {
+        SOCKADDR_HV *a = (SOCKADDR_HV *) addr;
+
+        wchar_t *guidStr;
+        if (UuidToStringW(&a->VmId, &guidStr) == RPC_S_OUT_OF_MEMORY) {
```

Comment:
I'm not a huge fan of testing for a specific error (nor am I a huge fan of trusting documented error codes, though in this case it checks out).

Could we perhaps capture the return value, check that it's not success first, and then raise a Windows error (PyErr_SetFromWindowsErr or something like that)? That will either transform the RPC_S_OUT_OF_MEMORY or include it with a message, and it's not like anyone can do much about it anyway.

Reply:
`PyErr_SetFromWindowsErr` -- neat! I was unaware of that API :)